lib/Runtime/Base/FunctionBody.h

```diff
@@ -1779,12 +1779,12 @@ namespace Js
         virtual uint GetShortDisplayNameOffset() const { return m_displayShortNameOffset; }
         LPCWSTR GetSourceInfo(int& lineNumber, int& columnNumber) const;
     private:
-        FieldWithBarrier(const byte*) m_functionBytes;
-        FieldWithBarrier(ByteCodeCache*) m_cache;
-        FieldWithBarrier(const char16 *) m_displayName;  // Optional name
-        FieldWithBarrier(uint) m_displayNameLength;
-        FieldWithBarrier(uint) m_displayShortNameOffset;
-        FieldWithBarrier(NativeModule *) m_nativeModule;
+        Field(const byte*) m_functionBytes;
+        Field(ByteCodeCache*) m_cache;
+        Field(const char16 *) m_displayName;  // Optional name
```

Comment:
this sometimes is recycler allocated, sometimes not, should use FieldWithBarrier.
check m_displayNameIsRecyclerAllocated and FunctionProxy::SetDisplayName() for detail